### PR TITLE
Easily embeddable static widgets

### LINKED ISSUE #425: Easily embeddable widgets, plan forward
After talking to @SylvainCorlay , him and I came up with this plan forward.  It involves cherry picking commits from the docs PR and one of his PRs:

- move web4 widget manager to jupyter-js-widgets
- use jupyter-js-widgets.embed-manager in web4 and update the description of web4 accordingly
- cherry-pick the new command to embed widget
- allow base-manager to have partial state updates
- mention adding base64 option in the command

### PR DESCRIPTION
**Pull request by jdfreder**
- [x] move web4 widget manager to jupyter-js-widgets  
 - [x] use jupyter-js-widgets.embed-manager in web4 and update the description of web4 accordingly  
 - [x] cherry-pick the new command to embed widget  
 - [x] allow base-manager to have partial state updates   
~~add base64 option in the embed command~~

closes #425 

@SylvainCorlay

### CODE REVIEW

**Comment by jdfreder** [comment]
Test failure looks legitimate ...  we'll have to investigate prior to merge.

**Comment by jdfreder** [comment]
@SylvainCorlay I just rebased this.  I've skipped `add base64 option in the embed command` for now, is that okay?

**Comment by jdfreder** [comment]
👋 good to go

**Comment by jdfreder** [comment]
@SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay @SylvainCorlay

File: jupyter-js-widgets/src/embed-webpack.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  // Assume that the current script tag is the last script tag, because the
+  // script tags are processed immediately as the browser traverses through the
+  // DOM.
+  var context = Array.prototype.slice.call(document.querySelectorAll('script'), -1)[0];
+  context.parentElement.insertBefore(container, context);
+
+  // Create and render the widget, replacing the loading view.
+  const widgetContainer = document.createElement('div');
+  widgetContainer.className = 'widget-area';
+  manager.display_widget_state(widgetStateObject, widgetContainer).then(() => {
```
Comment (SylvainCorlay) [inline jupyter-js-widgets/src/embed-webpack.js]: I don't think that we transpile the web4 example. (usage of lambda)

**Comment by jdfreder** [comment]
Just noticed the ipywidgets 6.x (master) branch is somewhat broken post jquery remove, something to keep in mind.  @SylvainCorlay @afshin @dwillmer

**Comment by SylvainCorlay** [comment]
> Just noticed the ipywidgets 6.x (master) branch is somewhat broken post jquery remove, something to keep in mind. @SylvainCorlay @afshin @dwillmer

A few things we noticed after a quick check:
 -  IntSlider does not work
 - the standard Button is gray
 - the controller widgets does not work
post jquery removal.

**Comment by afshin** [comment]
@SylvainCorlay @jdfreder @dwillmer My jQuery fix PR was to get `IntSlider` working again:
https://github.com/ipython/ipywidgets/pull/437

The standard button style is something I left alone figuring we'll have a style reboot post-Bootstrap, but I'm happy to add a default in tomorrow.

**Comment by afshin** [comment]
FYI, I *think* the Travis build fails because it's using a *very, very old* node version (`node v0.10.36`) which does not support `Promise`s and there is no polyfill being used, either. But this issue is in the examples, which is code I haven't touched in this PR (or at all).

**Pull request merged**